pkg/registry/types.go

```diff
@@ -114,6 +114,71 @@ type Annotations struct {
 	DefaultChannelName string `json:"operators.operatorframework.io.bundle.channel.default.v1" yaml:"operators.operatorframework.io.bundle.channel.default.v1"`
 }
 
+// DependenciesFile holds dependency information about a bundle
+type DependenciesFile struct {
+	// Dependencies is a list of dependencies for a given bundle
+	Dependencies []Dependency `json:"dependencies" yaml:"dependencies"`
+}
+
+// Dependencies is a list of dependencies for a given bundle
+type Dependency struct {
+	// The type of dependency. It can be `olm.package` for operator-version based
+	// dependency or `olm.gvk` for gvk based dependency. This field is required.
+	Type string `json:"type" yaml:"type"`
+
+	// The name of dependency such as 'etcd'. This field is required.
+	Name string `json:"name" yaml:"name"`
+
+	// The group of GVK based dependency.
+	Group string `json:"group,omitempty" yaml:"group"`
```

Comment:
Should we have explicit fields here for each dependency type? Can we just have a `dependencyIdentifier` as a raw json object or something similar so that we don't have to hardcode all of these and ignore certain fields depending on the type of dependency?

Reply:
I thought I removed `omitempty` part already (will do). I want a list of concrete fields that registry will parse and `opm` can validate. When the dependencies information are returned back along with api.Bundle, OLM simply retrieves the fields via .GetSomething() func instead of having to parse the JSON object.

Reply:
I'm not arguing that the input serialization of this object can't look like this (because that's what unioning these fields looks like on the yaml object), but it seems like we could translate it in the database. If for every single new type we add to the dependencies we have to add a set of fields, that database table will get really unwieldy really fast. Can we make it slightly more generic on the other end is what I'm really asking

Reply:
I think we either stick with explicit fields or a big JSON blob. Doing it half-half will only force us to do more DB migration like we do now to get more info inside CSV into separate fields. A big JSON blob will limit our ability to query dependencies using its fields. I think the existing fields should serve us well for foreseeable future. I can see one more field `provided` and that should be it. If we end up with adding more fields, we can always split the big tables into smaller tables that handle a specific dependency type.